Create qube_heatpump.markdown (#42924)
home-assistant/home-assistant.github.io: source/_integrations/qube_heatpump.markdown

---
title: Qube Heat Pump
description: Instructions on how to integrate Qube Heat Pump within Home Assistant.
ha_category:
  - HVAC
ha_release: '2025.2'
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@MattieGit'
ha_domain: qube_heatpump
ha_platforms:
  - binary_sensor
  - button
  - select
  - sensor
  - switch
---

> The **Qube Heat Pump** integration allows you to control and monitor

Reviewer: Please remove the line length limit and apply flowing text style throughout the page.
https://developers.home-assistant.io/docs/documenting/general-style-guide

[Qube](https://qube-renewables.com/) heat pumps via the Modbus TCP protocol.

## Configuration

To add the **Qube Heat Pump** integration to your Home Assistant instance, use
this My button:

<!-- markdownlint-disable-next-line line-length -->
[![Open your Home Assistant instance and start configuration](https://my.home-assistant.io/badges/config_flow_start.svg)](https://my.home-assistant.io/redirect/config_flow_start/?domain=qube_heatpump)

Manual steps:

1. Browse to your Home Assistant instance.
2. Go to **Settings > Devices & Services**.
3. In the bottom right corner, select the **Add Integration** button.
4. From the list, search for and select **"Qube Heat Pump"**.

> 5. Follow the instructions on screen.

Reviewer: ```suggestion
{% include integrations/config_flow.md %}
```
We have a predefined text element for this. See also:
https://developers.home-assistant.io/docs/documenting/create-page
https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L51

### Configuration Options

> 

Reviewer: ```suggestion
```

During the initial setup, you will be asked for:

> 

Reviewer: ```suggestion

{% configuration_basic %}
```

- **Host**: The IP address or hostname of your Qube Heat Pump

>   (default: `qube.local`).

Reviewer: ```suggestion
Host:
   description: "The IP address or hostname of your Qube Heat Pump (default: `qube.local`).
{% endconfiguration_basic %}
```
We use the configuration_basic tag for this.
https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L57C1-L57C26

> ### Options

Reviewer: ```suggestion
### Configuration options
```

After configuration, you can change the following settings by selecting
**Configure** on the integration entry:

> 

Reviewer: same here. Please wrap this into a configuration_basic block. You can find an example here:
https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L81-L86

- **Host**: Update the IP address if it changes.
- **Label**: An optional label to append to entity names (useful for
  differentiating multiple units).
- **Show label in name**: Toggle whether the label is included in the entity
  display name.
- **Friendly name language**: Select the language for entity names (`en` for
  English, `nl` for Dutch).

> ## Platforms

Reviewer: ```suggestion
## Supported functionality
```

The integration exposes various entities to monitor and control your heat pump.

> ### Sensor

Reviewer: ```suggestion
### Sensors
```

> 

Reviewer: please add the unit in addition to the sensor name, where applicable.

- **Temperatures**: Inlet, Outlet, Buffer, Outside, and Domestic Hot Water (DHW)
  temperatures.
- **Flow**: Current flow rate.
- **Power & Energy**:
  - **Standby power/energy**: Tracks the fixed 17W standby consumption.
  - **Total electric consumption**: Combined total valid for billing/tracking.
  - **SCOP (month)**: Monthly Seasonal Coefficient of Performance for CV
    (heating) and DHW, calculated from thermal output divided by electrical
    input.
- **Status**: Operation hours and system counters.

> ### Binary Sensor

Reviewer: ```suggestion
### Binary sensors
```

- **Components**: Status of the pump, compressor, and auxiliary systems.
- **I/O**: State of various inputs and outputs.
- **Alarms**: Active alarm states (e.g. *Flow CV*, *Supply too hot*).

### Select

- **SG Ready mode**: Consolidates the two SG Ready coil switches into a single
  readable selector:
  - **Off**: Normal operation.
  - **Block**: Heat pump operation blocked.
  - **Plus**: Regular heating curve, +1K room temp, tapwater day mode.
  - **Max**: Anti-legionella program once, surplus heating control, +1K room
    temp.

### Switch

- **Enable heating curve**: Toggles weather-compensated heating. When *Off*, the
  pump uses a fixed setpoint.
- **Activate central heating (Modbus Demand)**: Used for external thermostat
  control (see below).

> ## Thermostat Control

Reviewer: ```suggestion
## Thermostat control
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization), also in headings.

You can control the Qube heat pump using a virtual thermostat in Home Assistant.

1. **Enable Modbus Control**: Turn on the `Activate central heating` switch
   (and optionally `Activate DHW heating`).
2. **Disable Internal Thermostat**: On the physical heat pump controller,
   disable the "Linq" thermostat options for room temperature and domestic hot
   water.
   - *Note: Leaving Linq options enabled while using Modbus coils can lead to
     conflicting control.*
3. **Automation**: Use your Home Assistant automations or generic thermostat
   entity to toggle the demand switch based on your room temperature sensors.

### Button

- **Reload**: Refreshes all data from the heat pump immediately.

> ## Actions

Reviewer: Are actions already part of the initial core PR? If not, please remove all content that is not covered by the core PR.

Author: I have removed the action, as they were using switches which will not be part of the initial PR (only sensors). Thanks for the reminder.

### Action `qube_heatpump.write_register`

Write a value to a Modbus holding register exposed by the Qube heat pump. This
supports specific data types used by the device.

<!-- markdownlint-disable line-length -->
| Data attribute | Optional | Description |
| :--- | :--- | :--- |
| `address` | No | The Modbus holding register address to write (0-based). |
| `value` | No | The value to store in the register. |
| `data_type` | Yes | Register data type (e.g. `float32`). |
| `label` | Yes | Target a specific hub by its configured label (e.g. `qube1`) if multiple are present. |
<!-- markdownlint-enable line-length -->

> ## Multiple Hubs

Reviewer: ```suggestion
## Multiple hubs
```

If you have multiple Qube units:

> - Each hub receives a persistent label (e.g. `qube1`, `qube2`) configured in the

Reviewer: ```suggestion
- Each hub receives a persistent label (for example, `qube1`, `qube2`) configured in the
```
we prefer "for example", or "such as" over e.g., see also https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg

  Options flow.

> - Entity IDs are automatically prefixed (e.g.,

Reviewer: ```suggestion
- Entity IDs are automatically prefixed (for example,
```

>   `sensor.qube1_outlet_temperature`) to prevent collisions.

Reviewer: ```suggestion
  `sensor.qube1_outlet_temperature`) to prevent collisions.
  
## Removing the integration

This integration follows standard integration removal.

{% include integrations/remove_device_service.md %}
```
If there are additional steps required to remove the integration, please add them.

---
title: Qube Heat Pump
description: Instructions on how to integrate Qube Heat Pump within Home Assistant.
ha_category:
  - HVAC

> ha_release: '2025.2'

Reviewer: ```suggestion
ha_release: 2026.2
```

Author: Done!

ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@MattieGit'
ha_domain: qube_heatpump
ha_platforms:
  - sensor
---

The **Qube Heat Pump** {% term integration %} allows you to control and monitor [Qube](https://www.hr-energy.com/nl/pvt-systemen/onderdelen/qube-warmtepomp/) heat pumps via the Modbus TCP protocol.

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:

>   description: The IP address or hostname of your Qube Heat Pump (default: `qube.local`).

Reviewer: ```suggestion
  description: The IP address or hostname of your Qube Heat Pump (default is `qube.local`).
```

{% endconfiguration_basic %}

## Configuration options

After configuration, you can change the following settings by selecting **Configure** on the integration entry:

{% configuration_basic %}
Host:
  description: Update the IP address if it changes.
Label:
  description: An optional label to append to entity names (useful for differentiating multiple units).
{% endconfiguration_basic %}

## Supported functionality

The integration exposes various entities to monitor and control your heat pump.

### Sensor

> - **Temperatures**: inlet (°C), outlet (°C), buffer (°C), outside (°C), and domestic hot water (DHW) temperatures (°C).

Reviewer: the text in bold: are they UI strings? or titles?
We use bold to markup UI strings. If possible, avoid it to hightlight text.

Author: These are titles, so will remove the "bold" formatting.

- **Flow**: Current flow rate (L/min).
- **Power and energy**: standby power and total electric consumption (kWh).
- **Status**: Operation hours and system counters.

> ## Removing the integration

Reviewer: ```suggestion
## Removing the integration

```

> This integration follows standard integration removal.

Reviewer: ```suggestion
This integration follows standard integration removal.

```

---
title: Qube Heat Pump
description: Instructions on how to integrate Qube Heat Pump within Home Assistant.
ha_release: 2026.2

> ha_iot_class: Local Polling

Reviewer: ```suggestion
ha_iot_class: Local Polling
config_flow: true
```

---
title: Qube Heat Pump
description: Instructions on how to integrate Qube Heat Pump within Home Assistant.

> ha_release: 2025.3

Reviewer: ```suggestion
ha_release: 2026.2
```

ha_integration_type: hub
---

The **Qube Heat Pump** {% term integration %} allows you to control and monitor [Qube](https://www.hr-energy.com/nl/pvt-systemen/onderdelen/qube-warmtepomp/) heat pumps via the Modbus TCP protocol.

## Supported devices

The following devices are known to be supported by the integration:

- Qube heat pump

## Unsupported devices

The following devices are not supported by the integration:

- Qbooster heat pump (the predecessor of the Qube heat pump)

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:
  description: The IP address or hostname of your Qube Heat Pump (default is `qube.local`).
{% endconfiguration_basic %}

## Configuration options

After configuration, you can change the following settings by selecting **Configure** on the integration entry:

{% configuration_basic %}
Host:
  description: Update the IP address if it changes.
Label:
  description: An optional label to append to entity names (useful for differentiating multiple units).
{% endconfiguration_basic %}

## Supported functionality

The **Qube Heat Pump** integration exposes various entities to monitor and control your heat pump.

### Sensor

> - Temperatures: inlet (°C), outlet (°C), buffer (°C), outside (°C), central heating (CH) and domestic hot water (DHW) temperatures (°C).

Reviewer: especially for sensors where the name is not self-explanatory, we add a description. For example:
**Buffer** (°C): description of what a temperature buffer is.

Author: I have removed the reference to buffer and added "minimum and maximum setpoints" which is a better description.

- Flow: Current flow rate (L/min).

> - Power and energy: standby power and total electric consumption (kWh).

Reviewer: the unit of standby power is also kWh? Or is it W?

Author: It is W, indeed. Will update this in new commit. The energy sensors (in kWh) are not part of the initial PR for this new integration.

> - Status: Operation hours and system counters (h).

Reviewer: the system counter is in hours? What is a system counter?

Author: Only the operation hours are remaining, so will remove system counter reference.

---
title: Qube Heat Pump
description: Instructions on how to integrate Qube Heat Pump within Home Assistant.
ha_release: 2025.3
ha_iot_class: Local Polling
ha_codeowners:
  - '@MattieGit'
ha_domain: qube_heatpump
ha_integration_type: hub
---

<!--- The integration documentation template provides a documentation structure as well as some example content per section. The example content is meant for inspiration, it may not apply for your integration or will at least have to be adapted. -->

> <!--- Use this template together with the developer documentation, under [Documentation standard](https://developers.home-assistant.io/docs/documenting/standards) and the documentation rules of the [Integration Quality Scale](https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/). -->

Reviewer: ```suggestion
```

---
title: Qube Heat Pump
description: Instructions on how to integrate Qube Heat Pump within Home Assistant.

> ha_release: 2026.2

Reviewer: ```suggestion
ha_release: 2026.4
```
